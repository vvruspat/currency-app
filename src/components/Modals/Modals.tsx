import React, { useCallback, useState } from "react";
import {
  SelectModal,
  SelectModalProps,
} from "../uikit/Select/SelectModal/SelectModal";

export const ModalsContext = React.createContext<{
  setModal: (modal: ModalsProps | null) => void;
}>({
  setModal: (modal: ModalsProps | null) => {},
});

export enum MODALS {
  SELECT_MODAL = "select-modal",
  ERROR_MODAL = "error-modal",
  EXCHANGE_SUCCESS_MODAL = "exchange-success-modal",
}

// TODO: render and start modals from the component where it need

type ModalsProps = {
  modal: MODALS.SELECT_MODAL;
  props: SelectModalProps;
};
// | { modal: MODALS.ERROR_MODAL; props: /*ErrorModalProps*/ {} }
// | {
//     modal: MODALS.EXCHANGE_SUCCESS_MODAL;
//     props: /*ExchangeSuccessModalProps*/ {};
//   };

type ModalsRootProps = {};

export const ModalsRoot: React.FC<ModalsRootProps> = ({ children }) => {
  const [currentModal, setCurrentModal] = useState<ModalsProps | null>(null);

  const setModal = useCallback((modal: ModalsProps | null) => {
    setCurrentModal(modal);
  }, []);

  return (
    <ModalsContext.Provider value={{ setModal }}>
      {children}
      {/* <ErrorModal
        {...currentModalProps}
        show={currentModal === MODALS.ERROR_MODAL}
      /> */}
      <SelectModal
        {...currentModal?.props}
        show={currentModal?.modal === MODALS.SELECT_MODAL}
      />
      {/* <ExchangeSuccessModal
        {...currentModalProps}
        show={currentModal === MODALS.EXCHANGE_SUCCESS_MODAL}
      /> */}
    </ModalsContext.Provider>
  );
};
