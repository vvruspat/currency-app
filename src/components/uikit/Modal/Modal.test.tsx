import { render, screen } from "@testing-library/react";
import { Modal } from ".";
import { Title } from "..";
import { MODALS } from "../../Modals/Modals";

test("renders modal without header", () => {
  render(<Modal nav={MODALS.TEST_MODAL} data-testid="test-modal" />, {});
  const ModalElement = screen.getByTestId("test-modal");
  expect(ModalElement).toBeInTheDocument();
  expect(ModalElement).toMatchSnapshot();
});

test("renders modal with header", () => {
  render(
    <Modal
      nav={MODALS.TEST_MODAL}
      data-testid="test-modal"
      header={<Title data-testid="test-modal-header">Test header</Title>}
    />,
    {}
  );

  const ModalElement = screen.getByTestId("test-modal");
  expect(ModalElement).toBeInTheDocument();
  expect(ModalElement).toMatchSnapshot();

  const modalHeaderElement = screen.getByTestId("test-modal-header");
  expect(modalHeaderElement).toBeInTheDocument();
});
