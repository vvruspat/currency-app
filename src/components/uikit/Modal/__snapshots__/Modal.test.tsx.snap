// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders modal with header 1`] = `
<div
  class="modal"
  data-testid="test-modal"
>
  <header
    class="modal-header"
  >
    <div
      class="modal-header-content"
    >
      <div
        class="caption"
      >
        <div
          class="title"
          data-testid="test-modal-header"
        >
          Test header
        </div>
      </div>
    </div>
    <button
      class="icon-button modal-header-close-button"
    >
      <div
        class="icon-button-content"
      >
        <div
          class="icon-button-icon"
        >
          <svg>
            Cross16Icon.svg
          </svg>
        </div>
        <div
          class="icon-button-text"
        >
          Close
        </div>
      </div>
    </button>
  </header>
  <section
    class="modal-content"
  />
</div>
`;

exports[`renders modal without header 1`] = `
<div
  class="modal"
  data-testid="test-modal"
>
  <section
    class="modal-content"
  />
</div>
`;
